dask/dataframe/tests/test_io.py

```diff
@@ -684,14 +684,47 @@ def test_read_hdf():
         assert a.npartitions == 2
         assert a._known_dtype
 
-        tm.assert_frame_equal(a.compute(), df)
+        eq(a, df)
 
-        tm.assert_frame_equal(
-              dd.read_hdf(fn, '/data', chunksize=2, start=1, stop=3).compute(),
-              pd.read_hdf(fn, '/data', start=1, stop=3))
+        eq(dd.read_hdf(fn, '/data', chunksize=2, start=1, stop=3),
+           pd.read_hdf(fn, '/data', start=1, stop=3))
 
-        assert sorted(dd.read_hdf(fn, '/data').dask) == \
-               sorted(dd.read_hdf(fn, '/data').dask)
+        assert (sorted(dd.read_hdf(fn, '/data').dask) ==
+                sorted(dd.read_hdf(fn, '/data').dask))
+
+    with tmpfile('h5') as fn:
+        df.to_hdf(fn, '/data', format='table')
+        a = dd.read_hdf(fn, '/data', chunksize=2, sorted_index_column=True)
+        eq(a.npartitions, 2)
+        assert a._known_dtype
+        eq(a.divisions, (1., 3., 4.))
+
+        eq(a.compute(), df)
+
+        a = dd.read_hdf(fn, '/data', chunksize=2, start=1, stop=3,
+                        sorted_index_column=True)
+        b = pd.read_hdf(fn, '/data', start=1, stop=3)
+
+        eq(a.divisions, (2., 4.))
+
+        eq(a, b)
+
+    with tmpfile('h5') as fn:
+        df.to_hdf(fn, '/data', format='table')
+        a = dd.read_hdf(fn, '/data', chunksize=2, sorted_index_column='y')
+        eq(a.npartitions, 2)
```

Comment:
We generally only use `eq` to compare dasky things.  I recommend sticking to `assert x == y` for normal Python values.